tests/functional/dbt_runner/test_dbt_runner.py

```diff
@@ -35,3 +37,29 @@ def test_callbacks(self) -> None:
         # to have a project to run it and it will emit events
         dbt.invoke(["debug"])
         mock_callback.assert_called()
+
+    def test_invoke_override(self, project, dbt):
+        results, success = dbt.invoke(
+            ["run"],
+            log_format="json",
+            log_path="some_random_path",
+            version_check=False,
+            profile_name="some_random_profile_name",
+            target_dir="some_random_target_dir",
+        )
```

Comment:
It would probably be a good idea to add tests to assert which params take precedence, kwargs or flags `dbt.invoke(["--log-format=text", "run"], log_format="json")` .